narwhal/consensus/src/metrics.rs

```diff
@@ -23,6 +24,21 @@ pub struct ConsensusMetrics {
     pub recovered_consensus_output: IntCounter,
     /// The approximate size in memory (including heap allocations) of the Dag.
     pub dag_size_bytes: IntGauge,
+    /// The latency between two successful commit rounds
+    pub commit_rounds_latency: Histogram,
+    /// The number of certificates committed
+    /// per commit round
+    pub commit_depth: Histogram,
+    /// When a certificate is received on an odd round, we check
```

Comment:
The current implementation seems to increment the metric when `round % 2 == 0`? Also, how should we interpret this metric?

Reply:
> Also, how should we interpret this metric?

Leaders get elected for `even` rounds. The trigger actually is a certificate that is received for an `odd` round - so practically our leader election process is taking place always for the "previous" round.

This metric here will help us identify all the rounds on which we didn't manage to successfully elect a leader. What that does that practically mean? We "lost" a round on which we could commit certificates but we didn't and we have to wait until the next even round to commit. That is translated to delay.

I want to be able to track how frequently we are missing the opportunities (rounds) for commit and correlate with events (ex because we see nodes be down and not propose).

> The current implementation seems to increment the metric when round % 2 == 0?
Yes , as I said above we evaluate a leader when we are processing a certificate of `odd` round. However, I can't be sure that a leader has not been elected until we have moved to next round (thus we have received at least 2f +1 certificates on the current `odd`). Thus I'll need to increment the metric on the next `even` round